components/regionDropdowns/ChannelDropdown.tsx

```diff
@@ -0,0 +1,55 @@
+import { useRouter } from "next/router";
+import React from "react";
+
+import { useRegions } from "../RegionsProvider";
+import { BaseRegionsDropdown } from "./BaseRegionsDropdown";
+import { BaseRegionsDropdownItem } from "./BaseRegionsDropdownItem";
+
+interface DropdownOption {
+  label: string;
+  chosen: boolean;
+  channelSlug: string;
+}
+
+export function ChannelDropdown() {
+  const router = useRouter();
+  const { channels, currentChannel, setCurrentChannel, currentLocale } = useRegions();
+
+  const channelOptions: DropdownOption[] = channels.map((ch) => ({
+    label: ch.name,
+    chosen: ch.slug === currentChannel.slug,
+    channelSlug: ch.slug,
+  }));
+
+  const onChannelChange = async (channelSlug: string) => {
+    if (channelSlug === currentChannel.slug) {
+      return;
+    }
+    await setCurrentChannel(channelSlug);
+
+    // Update current URL to use the chosen channel
+    await router.push({
```

Comment:
```suggestion
   setCurrentChannel(channelSlug);

    // Update current URL to use the chosen channel
   router.push({
```
these functions are synchronous